.github/workflows/test-integration.yml

```diff
@@ -0,0 +1,53 @@
+name: test-integration
+on:
+  push:
+    branches: [main]
+  pull_request_review:
+    types: [submitted, dismissed]
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+*"
+
+jobs:
+  test-functional:
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '^1.17'
+
+    # set up PATH env based on OS
+    - name: add bin dir (unix)
+      if: runner.os != 'Windows'
+      run: echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+    - name: add bin dir (windows)
+      if: runner.os == 'Windows'
+      run: echo "${env:GITHUB_WORKSPACE}\bin" | Out-File -FilePath ${env:GITHUB_PATH} -Encoding utf8 -Append
+
+    # set up our binaries
+    - name: build binary to test
+      run: |
+        mkdir -p bin
+        go build -o bin
+        which hcdiag
+        hcdiag -version
+    - name: install softwares
+      run: |
+        go install github.com/gulducat/hashi-bin@main
+        cd bin
+        hashi-bin download consul latest
```

Comment:
this is a great point, and reaches into both the past and the future -- it might be nice to know the actual earliest versions of the products that hcdiag will work with, and some in between, and also latest.  definitely room for improvement here!